src/main/java/net/neoforged/neoforge/flag/RequiredFlagsCondition.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.flag;
+
+import com.mojang.serialization.MapCodec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import it.unimi.dsi.fastutil.objects.ReferenceOpenHashSet;
+import java.util.Collection;
+import java.util.Set;
+import java.util.stream.Collectors;
+import net.neoforged.neoforge.common.conditions.ICondition;
+import net.neoforged.neoforge.common.util.NeoForgeExtraCodecs;
+
+/**
+ * {@link ICondition Condition} used to conditionally load data files based on state of given set of {@link Flag Flags}.
+ * <p>
+ * All provided {@link #requiredFlags flags} must be enabled in order for data to pass this {@link ICondition condition}.
+ *
+ * @apiNote A {@code /reload} is be required to correctly reload data after toggling the state of a given {@link Flag}.
+ */
+public final class RequiredFlagsCondition implements ICondition {
+    public static final MapCodec<RequiredFlagsCondition> CODEC = RecordCodecBuilder.mapCodec(builder -> builder.group(
+            NeoForgeExtraCodecs.setOf(Flag.CODEC).fieldOf("flags").forGetter(condition -> condition.requiredFlags)).apply(builder, RequiredFlagsCondition::new));
+
+    private final Set<Flag> requiredFlags;
+
+    public RequiredFlagsCondition(Flag... requiredFlags) {
+        this.requiredFlags = new ReferenceOpenHashSet<>(requiredFlags);
+    }
+
+    private RequiredFlagsCondition(Collection<Flag> requiredFlags) {
+        this.requiredFlags = new ReferenceOpenHashSet<>(requiredFlags);
+    }
+
+    @Override
+    public boolean test(IContext context) {
```

Comment:
The context should provide access to the flag manager instead of having to use the global lookup.

Reply:
Looking into this I cant think of a good way to pass the flag manager instance from server to the context due to where it is being constructed (deep in `ReloadableServerResources`).

But even if I could this has also lead me to realize that the condition just flat out does not work during initial server startup, this is due to there being no server instance and potentially also level data not being loaded yet. Meaning we cannot lookup the flag manager or even if we could no flag states might have been loaded yet, which then causes everything to default to flags being disabled.

While I could potentially refactor where the flag manager is constructed server side, so that it is available before a server is spun up, there would still be the level data issue. Level data might not have been loaded yet meaning the saved flag states are not loaded and flags are assumed to be disabled.

So im at a bit of a loss on how to go about resolving this, my initial thoughts are I need to look into how vanilla loads their feature flags from disk but that looks to involve patching  `WorldDataConfiguration` and quite a few other places to load in the flags and pass them around everywhere, which just feels quite icky too me. Passing around this raw list of flags until eventually handing them off to the flag manager.

Reply:
This should now be working as intended and no longer requires global lookup.

Only global lookup left in is inside of `FlagElement`, see attached comment as to why.